test/test_nn.py

```diff
@@ -1180,14 +1180,39 @@ def compare_scaling(grads):
             for p, g in zip(l.parameters(), grads):
                 p.grad.data.copy_(g)
             norm_before = compute_norm(norm_type)
-            norm = clip_grad_norm(l.parameters(), max_norm, norm_type=norm_type)
+            norm = clip_grad_norm_(l.parameters(), max_norm, norm_type=norm_type)
             norm_after = compute_norm(norm_type)
             self.assertEqual(norm, norm_before)
             self.assertEqual(norm_before, norm_after)
             self.assertLessEqual(norm_after, max_norm)
             scale = compare_scaling(grads)
             self.assertEqual(scale, 1)
 
+    def test_clip_grad_value(self):
+        l = nn.Linear(10, 10)
+        clip_value = 2.5
+
+        grads = torch.arange(-50, 50).view(10, 10).div(5), torch.ones(10).mul(2)
+        for p, g in zip(l.parameters(), grads):
+            p._grad = Variable(g.clone().view_as(p.data))
+
+        clip_grad_value_(l.parameters(), clip_value)
+        for p in l.parameters():
+            self.assertLessEqual(p.grad.data.max(), clip_value)
+            self.assertGreaterEqual(p.grad.data.min(), -clip_value)
+
+        grads = torch.arange(-50, 50).view(10, 10).div(5), torch.ones(10).mul(2)
+        for i, (p, g) in enumerate(zip(l.parameters(), grads)):
+            if i == 0:
+                p._grad = Variable(g.clone().view_as(p.data))
+            else:
+                p._grad = None
+
+        clip_grad_value_(l.parameters(), clip_value)
+        for p in filter(lambda p: p.grad is not None, l.parameters()):
+            self.assertLessEqual(p.grad.data.max(), clip_value)
+            self.assertGreaterEqual(p.grad.data.min(), -clip_value)
```

Comment:
Can you please fold both of these cases into a single one? They look pretty much the same, only change them to be:
```python
grad_w, grad_b = ...
for grad_list in [[grad_w, grad_b], [grad_w, None]]:
    for p, g in zip(l.parameters(), grad_list):
        p._grad = g
    # do test here
```

Reply:
Yes, good point.